freelist.go

```diff
@@ -171,7 +172,9 @@ func (f *freelist) read(p *page) {
 	copy(f.ids, ids)
 
 	// Make sure they're sorted.
-	sort.Sort(pgids(f.ids))
+	if !pgids(f.ids).isSorted() {
```

Comment:
Have you benchmarked to confirm that it is faster on sorted data to first ask whether it is sorted? It is not obvious to me. Many sort algorithms run very quickly on already-sorted data, and when f.ids is large, touching all the data twice may be more expensive.